core/tsc.el

```diff
@@ -455,5 +456,22 @@ If a step cannot be followed, signal a `tsc--invalid-node-step' error."
         (setq this new-node)))
     this))
 
+(cl-defun tsc-build-parser-from-source (src-path &key generate dst-path)
+  "Build the tree-sitter grammar at SRC-PATH.
+
+If GENERATE is non-nil, the C parser will be generated from the `grammar.js'
+file in the repository root. In most cases, this is not necessary as the C
+parser is typically included in the repository.
+
+DST-PATH specifies whether the build result should be placed. If
+DST-PATH is unset, it will default to the value of
+`tree-sitter-langs-grammar-dir' if bound. Otherwise, an error will be raised."
+  (if-let ((dst-path (or dst-path (bound-and-true-p tree-sitter-langs-grammar-dir))))
```

Comment:
This feels a bit hacky to me, but I think users will want the build directory to default to the TSL grammar dir 99% of the time

Reply:
How about improving the situation with something like this:
``` lisp
(defvar tsc-dst-path-decider-func #'tsc-default-dst-path-decider
  "Function to decide the destination path for building the parser. 
Takes `src-path` and `dst-path` as arguments, should return the final `dst-path` to use.")

(defun tsc-default-dst-path-decider (src-path dst-path)
  "Default function for deciding destination path."
  (or dst-path (bound-and-true-p tree-sitter-langs-grammar-dir)))

(cl-defun tsc-build-parser-from-source (src-path &key generate dst-path)
;;;;
   (if-let ((dst-path (funcall tsc-dst-path-decider-func src-path dst-path)))